pkg/hive/hive_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	pb "github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/p2p/streamtest"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestInit(t *testing.T) {
+	logger := logging.New(ioutil.Discard, 0)
+	connectionManager := &ConnectionManagerMock{}
+	peerSuggester := &PeerSuggesterMock{}
+	addressFinder := &AddressFinderMock{}
+
+	// this is the receiving side
+	nodeReceiver := New(Options{
+		PeerSuggester: peerSuggester,
+		AddressFinder: addressFinder,
+		Logger:        logger,
+	})
+
+	// setup the stream recorder to record stream data
+	streamer := streamtest.New(
+		streamtest.WithProtocols(nodeReceiver.Protocol()),
+	)
+
+	nodeInit := New(Options{
+		Streamer:          streamer,
+		ConnectionManager: connectionManager,
+		Logger:            logger,
+	})
+
+	t.Run("OK", func(t *testing.T) {
+		suggesterPeers := make(map[int][]p2p.Peer)
```

Comment:
Sure, there is only one test-case implemented for now, this is why I kept it here. I agree about the table tests, but only if there happens to be a lot of different input/output combinations for the same flow. I am not sure if this is the case, I anticipate more combinations regarding different errors here, but I will give table tests a try as well when I start adding more tests. Makes sense?

Reply:
There is a map here, because as part of one tests there is a multiple, per bin, requests.

Reply:
@janos, @acud So, in which cases do you guys prefer table driven tests? I personally like to avoid them unless it's a perfect situation for them (table-like input data), but I also don't mind using them if you guys like them :) In this concrete example, my thinking is to add one happy path, and various error paths, which would be more readable without table tests, imo, but I might be wrong? 
Point number 2 holds true in either case.

Reply:
As there is only one test case, I think that it is ok to have it as it is, but I with Elad suggesting to create helper functions with some descriptive naming that would make the test more readable by less lines of code in the test function itself, as it is be a bit longer. Some suggestions `newMultiaddrs` to construct addresses in one line, and some for validating the peers from records that will hide away the protobuf parsing.

Reply:
Dealz, I'll do that for sure, just wanted to know about general view of table driven tests to know for the future as well, thanks!